### PR TITLE
Roll 44 firebase entry init

### LINKED ISSUE #73: ROLL-44 : Initialize new user firebase document
Check for the user's data in firebase.
If it exists, read from it to populate the profile and rollodex
Otherwise, create a new document and init to sign-up data

### PR DESCRIPTION
**Pull request by brickgriff**
Adds a blank firestore document to the `users` collection.

Bonus: updates the `static_users` collection methods for `user-list` and `user-details` pages.

https://github.com/bgriffiniv/rollme/issues/73

### CODE REVIEW

**Comment by brickgriff** [comment]
Will add notes later this evening.

File: src/app/pages/home/home.page.ts [hunk 1]
```diff
@@ -1,19 +1,50 @@
...
+          let navigationExtras: NavigationExtras = {
+                  state: {
+                    uid: currentUser.uid,
+                    firstName: '',
+                    lastName: '',
+                    email: '',
+                    mobile: ''
+                  }
+                };
+          this.router.navigateByUrl('/successful-signup', navigationExtras);
```
Comment (brickgriff) [inline src/app/pages/home/home.page.ts:45]: Yes I would prefer not to nest these calls but it's not so bad in this case. If there were more extensive branching logic then it would be harder to read. Will probably have to break it up for testability.

File: src/app/pages/home/profile/profile.page.ts [hunk 2]
```diff
@@ -66,7 +65,6 @@ export class ProfilePage implements OnInit {
           this.card.frontImg = this.router.getCurrentNavigation().extras.state.data;
       };
 
-      this.users = this.userService.getUsers();
       this.cards = this.cardService.getCards();
```
Comment (brickgriff) [inline src/app/pages/home/profile/profile.page.ts:68]: Will have to add a method to get cards belonging to a specific user (current auth user's uid)

Comment (brandlxee11) [reply src/app/pages/home/profile/profile.page.ts:68]: Yes exactly! Right there with you!

Comment (brandlxee11) [reply src/app/pages/home/profile/profile.page.ts]: Yes exactly!

File: src/app/pages/create-account/create-account.page.ts [hunk 3]
```diff
@@ -13,7 +13,6 @@ import { PasswordValidator } from '../../validators/password.validator';
 })
 export class CreateAccountPage implements OnInit {
 
-  firstName: '';
```
Comment (brandlxee11) [inline src/app/pages/create-account/create-account.page.ts:16]: So 'firstName' is no longer a standalone property but rather a value within the signupForm for 'currentUser', right?

File: src/app/pages/home/profile/profile.page.html [hunk 4]
```diff
@@ -6,15 +6,6 @@
         <img ng-show="isFrontCaptured" [src]="card.frontImg" alt=""/>
     </ion-card>
 
-    <ion-card button style="height:172px;" [routerLink]="['/user', user.id]" *ngFor="let user of (users | async)" lines="none">
```
Comment (brandlxee11) [inline src/app/pages/home/profile/profile.page.html]: I think this code bit will be useful when we test populate the rolodex for the infinite loop. You with me?

File: src/app/pages/home/profile/profile.page.html [hunk 5]
```diff
@@ -6,15 +6,6 @@
...
-      <ion-avatar class="image">
-        <img [src]="frontImg"/>
-      </ion-avatar>
-      <ion-card-header>
-        <ion-card-subtitle>{{ user.name }}</ion-card-subtitle>
-      </ion-card-header>
-    </ion-card>
-
```
Comment (brandlxee11) [inline src/app/pages/home/profile/profile.page.html:17]: I think this code bit will be useful when we test populate the rolodex for the infinite loop. You with me?

File: src/app/pages/user-list/user-list.page.html [hunk 6]
```diff
@@ -17,7 +17,7 @@
 
 
   <ion-list>
-    <ion-item button [routerLink]="['/user', user.id]" *ngFor="let user of (users | async)" lines="none">
+    <ion-item button [routerLink]="['/user', user.id]" *ngFor="let user of (users)" lines="none">
```
Comment (brandlxee11) [inline src/app/pages/user-list/user-list.page.html:20]: Why did you remove 'async'? This might be the start of the issue behind the newly created user not showing up on the user-list page.

File: src/app/pages/user-list/user-list.page.ts [hunk 7]
```diff
@@ -11,15 +11,22 @@ import { UserService, User } from 'src/app/services/user/user.service';
   styleUrls: ['./user-list.page.scss'],
 })
 export class UserListPage implements OnInit {
-  private users: Observable<User[]>;
```
Comment (brandlxee11) [inline src/app/pages/user-list/user-list.page.ts:14]: What difference does the 'Observable' make?

File: src/app/services/auth/auth.service.ts [hunk 8]
```diff
@@ -3,23 +3,32 @@ import { AngularFireAuth } from "angularfire2/auth";
...
 import * as firebase from 'firebase/app';
 
-//import { Router, NavigationExtras, ActivatedRoute } from '@angular/router';
-//import { FirebaseService } from 'src/app/services/firebase/firebase.service';
-
 @Injectable({
   providedIn: 'root'
 })
 export class AuthService {
-  user: Observable<firebase.User>;
```
Comment (brandlxee11) [inline src/app/services/auth/auth.service.ts:13]: So did you remove 'user' as a property altogether, or did you replace it with another?

File: src/app/services/user/user.service.ts [hunk 9]
```diff
@@ -18,46 +18,91 @@ export interface User {
...
+    .catch(e => {callback(e);});
   }
 
-  deleteUser(id: string): Promise<void> {
-    return this.userCollection.doc(id).delete();
+  deleteUser(user: User, callback) {
+    return this.userCollection.doc(user.id).delete()
+    .then(d => {callback(null,d);})
+    .catch(e => {callback(e);});
   }
```
Comment (brandlxee11) [inline src/app/services/user/user.service.ts:107]: What was the motivation behind refactoring the code here?

**Review by brandlxee11 — COMMENTED** [review]
Looks good! Let's just chat about the changes before merging.
